Source/WebGPU/WebGPU/Pipeline.mm

```diff
@@ -55,15 +57,73 @@
 
     auto prepareResult = WGSL::prepare(*ast, entryPoint, wgslPipelineLayout);
 
-    auto library = ShaderModule::createLibrary(device, prepareResult.msl, label);
-
     auto iterator = prepareResult.entryPoints.find(entryPoint);
     if (iterator == prepareResult.entryPoints.end())
         return std::nullopt;
 
     const auto& entryPointInformation = iterator->value;
```

Comment:
between line 64 and line 66 I think we need to populate the default values because they could be used for the compute workgroup size, for instance:
```
for (auto& kvp : entryPointInformation.specializationConstants) {
        auto& specializationConstant = kvp.value;
        if (!specializationConstant.defaultValue)
            continue;

        auto constantValue = WGSL::evaluate(*kvp.value.defaultValue, wgslConstantValues);
        auto addResult = wgslConstantValues.add(kvp.key, constantValue);
        ASSERT_UNUSED(addResult, addResult.isNewEntry);
    }
```